collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py

```diff
@@ -234,38 +234,37 @@ def update_ds(module, blade):
         attr = {}
         try:
             ds_now = blade.directory_services.list_directory_services(names=[module.params['dstype']]).items[0]
-            if module.params['dstype'] in ['management', 'smb']:
-                if module.params['dstype'] == 'nfs':
-                    if module.params['nis_servers']:
-                        if sorted(module.params['nis_servers']) != sorted(ds_now.nfs.nis_servers) or \
-                                module.params['nis_domain'] != ''.join(map(str, ds_now.nfs.nis_domains)):
-                            attr['nfs'] = {'nis_domains': [module.params['nis_domain']],
-                                           'nis_servers': module.params['nis_servers'][0:30]}
-                            mod_ds = True
-                    else:
-                        if module.params['uri']:
-                            if sorted(module.params['uri'][0:30]) != sorted(ds_now.uris):
-                                attr['uris'] = module.params['uri'][0:30]
-                                mod_ds = True
-                        if module.params['base_dn']:
-                            if module.params['base_dn'] != ds_now.base_dn:
-                                attr['base_dn'] = module.params['base_dn']
-                                mod_ds = True
-                        if module.params['bind_user']:
-                            if module.params['bind_user'] != ds_now.bind_user:
-                                attr['bind_user'] = module.params['bind_user']
-                                mod_ds = True
-                        if module.params['enable']:
-                            if module.params['enable'] != ds_now.enabled:
-                                attr['enabled'] = module.params['enable']
-                                mod_ds = True
-                        if module.params['bind_password']:
-                            attr['bind_password'] = module.params['bind_password']
-                            mod_ds = True
-                        if module.params['dstype'] == 'smb':
-                            if module.params['join_ou'] != ds_now.smb.join_ou:
-                                attr['smb'] = {'join_ou': module.params['join_ou']}
-                                mod_ds = True
+            if module.params['dstype'] == 'nfs':
+                if module.params['nis_servers']:
+                    if sorted(module.params['nis_servers']) != sorted(ds_now.nfs.nis_servers) or \
+                            module.params['nis_domain'] != ''.join(map(str, ds_now.nfs.nis_domains)):
+                        attr['nfs'] = {'nis_domains': [module.params['nis_domain']],
+                                       'nis_servers': module.params['nis_servers'][0:30]}
+                        mod_ds = True
+            else:
+                if module.params['uri']:
```

Comment:
Else only allowing SMB and Management Directory services.  It should allow NFS also if not configured for "NIS" .

Reply:
New push should resolve this oversight.